Content.Server/Speech/EntitySystems/IlleismAccentSystem.cs

```diff
@@ -0,0 +1,122 @@
+using System.Text.RegularExpressions;
+using Content.Server.Speech.Components;
+
+namespace Content.Server.Speech.EntitySystems;
+
+public sealed class IlleismAccentSystem : EntitySystem
+{
+    // I am going to Sec -> NAME is going to Sec
+    private static readonly Regex RegexIAmUpper = new(@"\bI\s*AM\b");
+    private static readonly Regex RegexIAmLower = new(@"\bi\s*am\b", RegexOptions.IgnoreCase);
+
+    // I have it -> NAME has it
+    private static readonly Regex RegexIHaveUpper = new(@"\bI\s*HAVE\b");
+    private static readonly Regex RegexIHaveLower = new(@"\bi\s*have\b", RegexOptions.IgnoreCase);
+
+    // I do! -> NAME does!
+    private static readonly Regex RegexIDoUpper = new(@"\bI\s*DO\b");
+    private static readonly Regex RegexIDoLower = new(@"\bi\s*do\b", RegexOptions.IgnoreCase);
+
+    // I don't! -> NAME doesn't!
+    private static readonly Regex RegexIDontUpper = new(@"\bI\s+DON'?T\b");
+    private static readonly Regex RegexIDontLower = new(@"\bi\s+don'?t\b", RegexOptions.IgnoreCase);
+
+    // I/Myself -> NAME
+    private static readonly Regex RegexMyselfUpper = new(@"\bMYSELF\b");
+    private static readonly Regex RegexI = new(@"\bI\b|\bmyself\b", RegexOptions.IgnoreCase);
+
+    // Me -> NAME
+    private static readonly Regex RegexMeUpper = new(@"\bME\b");
+    private static readonly Regex RegexMeLower = new(@"\bme\b", RegexOptions.IgnoreCase);
+
+    // My crowbar -> NAME's crowbar
+    // That's mine! -> That's NAME's
+    private static readonly Regex RegexMyUpper = new(@"\bMY\b|\bMINE\b");
+    private static readonly Regex RegexMyLower = new(@"\bmy\b|\bmine\b", RegexOptions.IgnoreCase);
+
+    // I'll do it -> NAME'll do it
+    private static readonly Regex RegexIllUpper = new(@"\bI'?LL\b");
```

Comment:
It might be better to use [source-generated regex](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators?pivots=dotnet-8-0) but i'm not sure maintainers will be happy about it - it is not used in code base currently.

Reply:
Yeah, I based my regex strings off of regex strings in other accents. I assume the maintainers would want to keep it standard across the accent system. Maybe a future refactoring of the accent system could improve the regex strings?